### ISSUE TITLE
oauth2

### ISSUE THREAD

**Issue by derrick56007** [stevenroose/dart-reddit#1]
Hi, I have never used oauth2 before but would like user-enabled auth. Could you make an example of the implementation or point me in the right direction? Thanks

**Comment by stevenroose** [comment]
Most information I can provide you is in the README.
- You need to register you app with Reddit: https://www.reddit.com/prefs/apps
- You will get a client id and a secret that you have to pass to the library using [`Reddit.authSetup`](http://www.dartdocs.org/documentation/reddit/0.3.1/index.html#reddit/reddit.Reddit@id_authSetup).
- Then you request an authorization URL using `Reddit.authUrl`, passing the URL where you want to receive the access token from the user. (There are some methods to get that token into your window, look them up it's general OAuth flow.)
- Pass the token, or just the response from the auth server (the request made to your redirect url) using `Reddit.authFinish`.

Done :)

**Comment by derrick56007** [comment]
Thanks!

**Comment by stevenroose** [comment]
As you can see, the library is fairly new, so if you run in any problems, just let me know, I'll get it fixed asap. 

I haven't been able to test the user auth flow myself yet, so you'll be the first one to use it :)

(Also: Is there an end in the DartRogue maze? I had 2 keys but couldnt find a way out :D)

**Comment by derrick56007** [comment]
I was able to get the auth running but haven't been able to test out all the queries and listings (I'll definitely tell you if i run into any complications). However I think this is a really useful library for dart devs and I'm hoping to use it to make the reddit interface a little more user friendly.

As for DartRogue (which I made to learn some dart): I haven't been working on it recently (levels and bosses haven't been added in) but I hope to work on it this coming summer!
